src/julia_internal.h

```diff
@@ -711,6 +710,45 @@ STATIC_INLINE char *jl_copy_str(char **to, const char *from)
 // Returns time in nanosec
 JL_DLLEXPORT uint64_t jl_hrtime(void);
 
+#ifdef JULIA_ENABLE_PARTR
+// congruential random number generator
+STATIC_INLINE void seed_cong(uint64_t *seed)
+{
+    *seed = jl_hrtime();
+}
+STATIC_INLINE void unbias_cong(uint64_t max, uint64_t *unbias)
+{
+    *unbias = UINT64_MAX - ((UINT64_MAX % max)+1);
+}
+STATIC_INLINE uint64_t cong(uint64_t max, uint64_t unbias, uint64_t *seed)
+{
+    while ((*seed = 69069 * (*seed) + 362437) > unbias)
+        ;
```

Comment:
Semicolon on separate line as a ghost of the empty loop body or unintentional extra space?